src/error.rs

```diff
@@ -0,0 +1,209 @@
+pub type BoxedError = Box<dyn std::error::Error>;
+pub type Result<T> = core::result::Result<T, BoxedError>;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(
+        "{{\
+            \"variant\":\"Matter\",\
+            \"details\":{0}\
+        }}"
+    )]
```

Comment:
The `json!` macro doesn't work here 😞 so this is a bit fragile. I'll add tests for these errors perhaps, that try to parse an instantiated error's json output string using `serde_json` or `json`.

Reply:
Tests implemented.